src/process.cpp

```diff
@@ -698,7 +714,11 @@ namespace proc {
     auto proc_opt = proc::parse(file_name);
```

Comment:
I strongly suggest to change signature of `std::optional<proc::proc_t> parse(const std::string &file_name)` to `std::optional<std:tuple<std::vector<ctx_t>, boost::process::v1::environment>> parse(const std::string &file_name)` and also remove `proc_t` constructor as it is another disaster waiting to happen (don't forget to initialize `int _app_id{0};`)